File: client-clojure/src/sirqul_io_t_platform/specs/address_response.clj
(ns sirqul-io-t-platform.specs.address-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def address-response-data
  {
   (ds/opt :streetAddress) string?
   (ds/opt :streetAddress2) string?
   (ds/opt :city) string?
   (ds/opt :stateCode) string?
   (ds/opt :state) string?
   (ds/opt :postalCode) string?
   (ds/opt :countryCode) string?
   (ds/opt :country) string?
   (ds/opt :display) string?
   })

(def address-response-spec
  (ds/spec
    {:name ::address-response
     :spec address-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/AddressResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import AddressResponse._

case class AddressResponse (
  streetAddress: Option[String],
streetAddress2: Option[String],
city: Option[String],
stateCode: Option[String],
state: Option[String],
postalCode: Option[String],
countryCode: Option[String],
country: Option[String],
display: Option[String])

object AddressResponse {
  import DateTimeCodecs._

  implicit val AddressResponseCodecJson: CodecJson[AddressResponse] = CodecJson.derive[AddressResponse]
  implicit val AddressResponseDecoder: EntityDecoder[AddressResponse] = jsonOf[AddressResponse]
  implicit val AddressResponseEncoder: EntityEncoder[AddressResponse] = jsonEncoderOf[AddressResponse]
}
